### ISSUE TITLE
Projects do not properly keep their parent project

### ISSUE THREAD

**Issue by SevenSoulsss** [go-vikunja/frontend#139]
### Description

The projects do not correctly keep their parent project in the display while they are well defined in database, example, if I define the project "Hatypik" to go into the parent project "Twitch" :
![image](https://github.com/go-vikunja/frontend/assets/46326574/9f5712f6-2f22-4da3-b244-be05a36c6a6e)

 it goes correctly
![image2](https://github.com/go-vikunja/frontend/assets/46326574/31ad5243-918d-40c3-b776-8b8887aed344)

But when I refresh the page, it does not retain its position
![image3](https://github.com/go-vikunja/frontend/assets/46326574/a9ab68a2-02c3-4a47-9bb4-c1598937e8a8)

I can not reproduce the bug on the demo version of vikunja, would the bug come from the stable version ?

No error on my server by putting the logs in debug mode (including in SQL queries in debug)

### Vikunja Frontend Version

0.22.0

### Vikunja API Version

0.22.0

### Browser and version

Google Chrome 120.0.6099.131

### Can you reproduce the bug on the Vikunja demo site?

No

**Comment by vladimiroff** [comment]
This started to happen on my instance only after I began to use Tasks.org via CalDAV. Even though in the project's settings the parent is correct, only visually displays above as a root project in the sidebar.

**Comment by ozuzo** [comment]
Seeing something similar. First thought it might be related to me bulk updating a couple of things in the database, but I've also seen it happen when I was sure only modifications were made through the frontend. Both frontend/api are 0.22.0.

With
![image](https://github.com/go-vikunja/frontend/assets/5175516/e2bf1e08-cc78-4cc4-bc38-4fff5bb5a723)
in the database, I see:
![image](https://github.com/go-vikunja/frontend/assets/5175516/42483b89-795e-4751-8c47-2d88b8f098de)
in the browser.

I think it might actually be an API issue - the /projects API call returns a 0 parent_project_id for most (but not all?) projects (though it is max 1 per "parent"):
![image](https://github.com/go-vikunja/frontend/assets/5175516/aeb9406a-a38e-4411-a396-639d3cc364d3)

**Comment by kolaente** [comment]
Thanks for the additional details.
This definitely looks like an api issue.

**Comment by kolaente** [comment]
This should now be fixed in https://github.com/go-vikunja/api/commit/96b4f2d1dbd6442fab0df2906d5640fc7193ba2e. Please check with the latest unstable build.

**Comment by ozuzo** [comment]
Before going crazy - did something change in the handling of logins between 22.0 and 22.1? I updated our version to test, and can't seem to log in anymore wit my browser-saved credentials? Getting a 1011 error on the /login call...

To elaborate: I'm running Vikunja on a shared server where the API port is not exposed, so I added a proxy in Apache to redirect /api calls to the backend running on its own port, until this morning that worked perfectly :-)

**Comment by kolaente** [comment]
@ozuzo No changes I can recall from the top of my head. Please open a new issue with this.
